models/image_classification_mobilenet/mobilenet.cpp

```diff
@@ -0,0 +1,1126 @@
+#include <opencv2/opencv.hpp>
+
+using namespace std;
+using namespace cv;
+using namespace dnn;
+
+vector< pair<dnn::Backend, dnn::Target> > backendTargetPairs = {
+        std::make_pair<dnn::Backend, dnn::Target>(dnn::DNN_BACKEND_OPENCV, dnn::DNN_TARGET_CPU),
+        std::make_pair<dnn::Backend, dnn::Target>(dnn::DNN_BACKEND_CUDA, dnn::DNN_TARGET_CUDA),
+        std::make_pair<dnn::Backend, dnn::Target>(dnn::DNN_BACKEND_CUDA, dnn::DNN_TARGET_CUDA_FP16),
+        std::make_pair<dnn::Backend, dnn::Target>(dnn::DNN_BACKEND_TIMVX, dnn::DNN_TARGET_NPU),
+        std::make_pair<dnn::Backend, dnn::Target>(dnn::DNN_BACKEND_CANN, dnn::DNN_TARGET_NPU)};
+        
+
+std::string keys =
+"{ help  h          | | Print help message. }"
+"{ model m             | image_classification_mobilenetv1_2022apr.onnx | Usage: Set model type, defaults to image_classification_mobilenetv1_2022apr.onnx (v1) }"
+"{ input i          | | Path to input image or video file. Skip this argument to capture frames from a camera.}"
+"{ initial_width    | 0 | Preprocess input image by initial resizing to a specific width.}"
+"{ initial_height   | 0 | Preprocess input image by initial resizing to a specific height.}"
+"{ rgb              | true | swap R and B plane.}"
+"{ crop             | false | Preprocess input image by center cropping.}"
+"{ backend bt         | 0 | Choose one of computation backends: "
+"0: (default) OpenCV implementation + CPU,"
+"1: CUDA + GPU (CUDA),"
+"2: CUDA + GPU (CUDA FP16),"
+"3: TIM-VX + NPU,"
+"4: CANN + NPU}";
+
+using namespace std;
+using namespace cv;
+using namespace dnn;
+
+vector<string> getLabelImagenet1k();
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+
+    parser.about("Use this script to run classification deep learning networks in opencv Zoo  using OpenCV.");
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    int rszWidth = parser.get<int>("initial_width");
+    int rszHeight = parser.get<int>("initial_height");
+    bool swapRB = parser.get<bool>("rgb");
+    bool crop = parser.get<bool>("crop");
+    String model = samples::findFile(parser.get<String>("model"));
+    int backendTargetid = parser.get<int>("backend");
+ 
+    if (model.empty())
+    {
+        CV_Error(Error::StsError, "Model file " + model + " not found");
+    }
+    vector<string> labels =  getLabelImagenet1k();
+
+    Net net = readNet(model);
+    net.setPreferableBackend(backendTargetPairs[backendTargetid].first);
+    net.setPreferableTarget(backendTargetPairs[backendTargetid].second);
+    //! [Open a video file or an image file or a camera stream]
+    VideoCapture cap;
+    if (parser.has("input"))
+        cap.open(parser.get<String>("input"));
+    else
+        cap.open(0, CAP_DSHOW);
```

Comment:
My webcam cannot be open with `CAP_DSHOW`. How about removing it and using just `cap.open(0)` instead?